examples/deutsch.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Demonstrates Deutsch's algorithm.
+
+Deutsch's algorithm is one of the simplest demonstrations of quantum parallelism
+and interference. It takes a black-box oracle implementing a Boolean function
+f(x), and determines whether f(0) and f(1) have the same parity using just one
+query.  This version of Deutsch's algorithm is a simplified and improved version
+from Nielsen and Chuang's textbook.
+
+=== REFERENCE ===
+
+Deutsch, David. "Quantum theory, the Church-Turing Principle and the universal
```

Comment:
Deutch's paper doesn't discuss the algorithm in terms of circuits which is what this implementation is based on. I suggest removing this even though this is where the algorithm originated.

Reply:
I think having the original reference is a good idea, but I wouldn't put it first.

Reply:
Okay, I've put the Wikipedia link first and the original paper second.